test/junit/scala/util/matching/RegexTest.scala

```diff
@@ -172,4 +172,20 @@ class RegexTest {
       assertEquals("aaaaa", aes)
     }
   }
+
+  @Test def `t10827 matches method`(): Unit = {
+    val r = """\d+""".r
+    assertTrue(r.matches("500"))
+    assertFalse(r.matches("foo"))
+    assertFalse(r.matches("123 123"))
+    assertFalse(r.matches("foo2"))
+  }
+
+  @Test def `t10827 matches method for unanchored Regex`(): Unit = {
+    val r = """\d+""".r.unanchored
+    assertTrue(r.matches("500"))
+    assertFalse(r.matches("abc"))
+    assertTrue(r.matches("123 123"))
```

Comment:
The Java API distinguishes matches (entire input), looking at (prefix) and find (anywhere). I would give this usage a second look. (It does correspond to what you get in a `match` expression from the extractor, i.e. `find` behavior.)

Reply:
I was already uncomfortable with it, but wanted to maintain parity with the extractor to avoid confusion since people will expect `r.matches(s)` iff `s match { case r() => true case _ => false }`

Taking a second look at the `unanchored` docs it specifically says it only affects extractor behavior, so that makes it pretty clear that it should not affect `matches`. I will adjust the PR.

Reply:
Reverted to original behavior after discussion

Reply:
Thanks.